src/ng/compile.js

```diff
@@ -3131,7 +3131,13 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
               childBoundTranscludeFn);
           }
           linkQueue = null;
-        }).catch(noop);
+        }).
+        catch(function(error) {
+          if (error instanceof Error) {
+            $exceptionHandler(error);
+          }
+        }).
+        catch(noop);
```

Comment:
Depends on what the `$exceptionHandler` does. Throwing from the `$exceptionHandler` (as happens in tests) results in a Possibly Unhandled Rejection error.